spec/models/site_status_spec.rb

```diff
@@ -38,6 +38,179 @@
     it { should be_audited.associated_with(:course) }
   end
 
+  describe 'creation' do
+    context 'when course has a running site' do
+      let(:site1)       { create(:site) }
```

Comment:
`create(:site) `

is flawed 

```ruby
 let(:site1)       { create(:site) } # site1.provider.id == 1
 let(:site2)       { create(:site) }  # site1.provider.id == 2

```

`site` model has a validate method that sets the `code` based on the associated `provider`

`code` needs to be unique for the `provider`

see
https://github.com/DFE-Digital/manage-courses-backend/blob/38c05dadca27976919a5271797f9ddd1736bbf6b/app/models/site.rb#L46



maybe something like

```ruby
 let(:provider)  { build(:provider, sites: [site1, site2]) } 
 let(:site1)       { build(:site) } 
 let(:site2)       { build(:site) } 
 let(:site_status1) { build(:site_status, :running, site: site1) }
 let(:course)     {create(:course, provider: provider, site_statuses: [site_status1])}

```

please try to defer creation at the latest point in time and only have one `create` method call

Otherwise its ends up like https://github.com/DFE-Digital/manage-courses-backend/pull/472 for the pain factory